### PR TITLE
Make maximum subscription queue length customizable

### PR DESCRIPTION
**Pull request by bbenligiray**
I'm opening it as a draft PR because I didn't update the docs yet but feel free to check it out

### CODE REVIEW

File: contracts/api3-server-v1/Api3Market.sol [hunk 1]
```diff
@@ -121,7 +119,17 @@ contract Api3Market is HashRegistry, ExtendedSelfMulticall, IApi3Market {
...
     /// @param proxyFactory_ ProxyFactory contract address
-    constructor(address owner_, address proxyFactory_) HashRegistry(owner_) {
+    /// @param maximumSubscriptionQueueLength_ Maximum subscription queue
+    /// length
+    constructor(
+        address owner_,
+        address proxyFactory_,
+        uint256 maximumSubscriptionQueueLength_
+    ) HashRegistry(owner_) {
+        require(
```
Comment (bbenligiray) [inline contracts/api3-server-v1/Api3Market.sol:129]: Maybe also enforce 10 as the maximum value?

File: deploy/1_deploy.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@ import type { OwnableCallForwarder, ProxyFactory } from '../src/index';
 module.exports = async () => {
   const { deploy, log } = deployments;
   const [deployer] = await ethers.getSigners();
+  const MAXIMUM_SUBSCRIPTION_QUEUE_LENGTH = 10;
```
Comment (bbenligiray) [inline deploy/1_deploy.ts:14]: !

File: contracts/api3-server-v1/Api3Market.sol [hunk 3]
```diff
@@ -55,19 +55,6 @@ contract Api3Market is HashRegistry, ExtendedSelfMulticall, IApi3Market {
...
 
-    /// @notice Maximum subscription queue length for a dAPI
-    /// @dev Some functionality in this contract requires to iterate through
-    /// the entire subscription queue for a dAPI, and the queue length is
-    /// limited to prevent this process from being bloated. Considering that
-    /// each item in the subscription queue has unique update parameters, the
-    /// length of the subscription queue is also limited by the number of
-    /// unique update parameters offered in the dAPI pricing Merkle tree. For
-    /// reference, at the time this contract is implemented, the API3 Market
-    /// offers 4 update parameter options, and this number is not expected to
```
Comment (bbenligiray) [inline contracts/api3-server-v1/Api3Market.sol:66]: We already want to do 5% (paired with OEV) so this is out the window

**Comment by bbenligiray** [comment]
This PR will help me with https://github.com/api3dao/tasks/issues/626 too because I will run the exhaustive tests with a queue length of 3 (as the logic that works for 3 should also work for 5 or 10)

File: contracts/api3-server-v1/Api3Market.sol (hunk 1 above)
Comment (acenolaza) [reply contracts/api3-server-v1/Api3Market.sol:129]: I don't really see the value in locking ourselves (the only ones deploying this contract) with a fixed value other than us making a mistake. I would prefer to just add more details to the `@param` comment

**Review by acenolaza — APPROVED** [review]

File: contracts/api3-server-v1/Api3Market.sol (hunk 1 above)
Comment (alikonuk1) [reply contracts/api3-server-v1/Api3Market.sol:129]: I also agree with Santiago here.

**Review by alikonuk1 — APPROVED** [review]

File: contracts/api3-server-v1/Api3Market.sol (hunk 1 above)
Comment (bbenligiray) [reply contracts/api3-server-v1/Api3Market.sol:129]: Yeah I agree, which is why I omitted it in the first place

**Pull request merged**